### PR TITLE
feat(domains-students): [skp-119]create student delete endpoint

### PR DESCRIPTION
**Pull request by aurimas12**

### CODE REVIEW

File: apps/api/src/app/models/students/student.service.ts [hunk 1]
```diff
@@ -4,12 +4,18 @@ import { Model, Types } from 'mongoose';
...
 export class StudentModelService {
   @InjectModel(ViewModels.STUDENTS) public model!: Model<StudentViewModel>;
 
   public async create(data: StudentCreated): Promise<void> {
+    console.log('model-service');
     await this.model.findOneAndUpdate({ _id: Types.ObjectId(data.id) }, data.student, { upsert: true })
   }
+
+  async delete({ id }: DeleteStudentPayloadDto) {
+    await this.model.deleteOne({ _id: id });
```
Comment (Gardemarin) [inline apps/api/src/app/models/students/student.service.ts:19]: Convert id back to the Types.ObjectId()

File: apps/api/src/app/view/students/handlers/created-student.handler.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ export class StudentCreatedHandler implements IEventHandler<StudentCreated>{
     @Inject() private readonly _studentModelService: StudentModelService;
     public async handle(event: StudentCreated) {
         await this._studentModelService.create(event)
+        console.log('views-handler');
```
Comment (Gardemarin) [inline apps/api/src/app/view/students/handlers/created-student.handler.ts:12]: Delete all console.log()

File: apps/api/src/app/domains/students/students.controller.ts [hunk 3]
```diff
@@ -11,6 +12,12 @@ export class StudentsController {
 
     @Post()
     async createStudent(@Body() payload: CreateStudentPayloadDto) {
+        console.log('domains-controller');
         return this._commandBus.execute(new CreateStudent(payload))
     }
+
+    @Post(':id/delete')
+    async deleteStudent(@Param('id') id: string) {
```
Comment (Gardemarin) [inline apps/api/src/app/domains/students/students.controller.ts:20]: use only `delete` (you are inside student controller).

File: apps/api/src/app/domains/students/students.controller.ts [hunk 4]
```diff
@@ -11,6 +12,12 @@ export class StudentsController {
 
     @Post()
     async createStudent(@Body() payload: CreateStudentPayloadDto) {
```
Comment (Gardemarin) [inline apps/api/src/app/domains/students/students.controller.ts:14]: Use only create (you are inside StudentsController).

File: apps/api/src/app/domains/students/students.controller.ts [hunk 5]
```diff
@@ -10,7 +11,13 @@ export class StudentsController {
...
     @Post()
-    async createStudent(@Body() payload: CreateStudentPayloadDto) {
+    async create(@Body() payload: CreateStudentPayloadDto) {
+        console.log('domains-controller');
         return this._commandBus.execute(new CreateStudent(payload))
     }
+
+    @Post(':id/delete')
+    async delete(@Param('id') id: string) {
+        return this._commandBus.execute(new DeleteStudent({ id }))
```
Comment (doncevas) [inline apps/api/src/app/domains/students/students.controller.ts:21]: set only id without object

File: apps/api/src/app/models/students/dto/delete-student-payload.dto.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { DeleteStudent } from '../../../models/students/student.interface';
+
+export class DeleteStudentPayloadDto implements DeleteStudent {
```
Comment (doncevas) [inline apps/api/src/app/models/students/dto/delete-student-payload.dto.ts:4]: remove this dto

File: apps/api/src/app/domains/students/commands/delete-student.command.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import { ICommandHandler, CommandHandler, EventPublisher } from '@nestjs/cqrs';
+import { Inject } from '@nestjs/common';
+import { StudentAggregate } from '../students.aggregate';
+import { StudentDeleted } from '../events/student-deleted.event';
+import { DeleteStudentPayloadDto } from '../../../models/students/dto/delete-student-payload.dto';
+
+export class DeleteStudent {
+    constructor(public data: DeleteStudentPayloadDto) {
```
Comment (doncevas) [inline apps/api/src/app/domains/students/commands/delete-student.command.ts:8]: data change to id: string

File: apps/api/src/app/domains/students/commands/delete-student.command.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    }
+};
+
+@CommandHandler(DeleteStudent)
+export class DeleteStudentHandler implements ICommandHandler<DeleteStudent> {
+    @Inject() private readonly _publisher: EventPublisher;
+    async execute({ data }: DeleteStudent) {
```
Comment (doncevas) [inline apps/api/src/app/domains/students/commands/delete-student.command.ts:15]: data change to id

File: apps/api/src/app/models/students/student.service.ts [hunk 9]
```diff
@@ -12,4 +14,8 @@ export class StudentModelService {
   public async create(data: StudentCreated): Promise<void> {
     await this.model.findOneAndUpdate({ _id: Types.ObjectId(data.id) }, data.student, { upsert: true })
   }
+
+  async delete({ id }: DeleteStudentPayloadDto) {
```
Comment (doncevas) [inline apps/api/src/app/models/students/student.service.ts:18]: only id: string

File: apps/api/src/app/models/students/student.interface.ts [hunk 10]
```diff
@@ -14,4 +14,7 @@ export interface CreateStudent {
   email: string,
 }
 
+export interface DeleteStudent {
```
Comment (doncevas) [inline apps/api/src/app/models/students/student.interface.ts:17]: remove this interface

**Pull request merged**